### PR TITLE
TELFE-222 Search view user profile

### PR DESCRIPTION
**Pull request by dptelicent**
Ticket: https://telicent.atlassian.net/browse/TELFE-221

### Goal

Given a user is viewing the Search app

and /whoami is configured in the deployment

When a user looks at the top right of the screen

Then the user will see their user details in their user profile component as shown in the design.

and then app switcher will now be on the top left of the screen

and still work

if /whoami is not configured

Then the user will not see their user profile component on the top right of the screen.

### Work Completed

- Username width in top right reduced in order not to overlap the app branding.
- Clicking anywhere in the username/icon/chevron now shows the user profile.

### Testing Method
- 🎗️ [storybook](https://telicent-oss.github.io/telicent-ds/TELFE-222-Search-View-user-profile/?path=/docs/data-display-user-profile-user-profile-content--docs)  created
- Existing tests.

### Engineering Notes 

Required for https://github.com/Telicent-io/telicent-search/pull/646 and https://github.com/Telicent-io/telicent-graph/pull/511.

---

<details>
  <summary><strong>🕵️ Review Guidance</strong></summary>

---

General guidance

- Generally, approve a PR if it makes the system better, even if it's not perfect. — [Google: The Standard of Code Review](https://google.github.io/eng-practices/review/reviewer/standard.html)
- The aim of both PR AUTHOR and PR REVIEWER is to get the code merged
- Aim for consensus, defined as _everyone can live with the outcome_

For PR REVIEWER:

1. Read the ticket & description
2. [Review the code](https://google.github.io/eng-practices/review/reviewer/looking-for.html)
3. Request any changes that are essential.
4. For non-essential comments:
   - Use prefixes, e.g. "**nit:** change to Pascal case"
     - **nit:** small, non-essential change
     - **obs:** just an observation, doesn't affect the PR
     - **idea:** a suggestion to think about
     - **q:** questions
   - Use modifiers, e.g. "**obs**`[pr-owner-resolve]`: Jim is also editing this file"
     - `pr-author-resolve` PR author can resolve after reading
     - `pr-author-delete` (rare) delete after reading to avoid confusion
5. try to add _at least_ a helpful comment per ~500 lines; less if the PR is already busy

For PR AUTHOR:

1. Aim for enough detail in PR description for things to go smoothly
2. After requested changes, [re-request a review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/about-pull-request-reviews#re-requesting-a-review) (so the PR shows up in [reviews-requested:@me](https://github.com/pulls?q=is%3Apr+is%3Aopen+archived%3Afalse+sort%3Aupdated-desc+review-requested%3A%40me+))

</details>

### CODE REVIEW

**Review by ACoolmanTelicent — APPROVED** [review]
Seems alright in storybook
https://telicent-oss.github.io/telicent-ds/TELFE-222-Search-View-user-profile/?path=/docs/data-display-user-profile-user-profile-content--docs

**Review by ACoolmanTelicent — APPROVED** [review]
Seems alright in storybook
https://telicent-oss.github.io/telicent-ds/TELFE-222-Search-View-user-profile/?path=/docs/data-display-user-profile-user-profile-content--docs

**Review by AlecsMissangu — APPROVED** [review]

**Review by AlecsMissangu — CHANGES_REQUESTED** [review]
The pop up seems to be opening in the incorrect place here
https://telicent-oss.github.io/telicent-ds/TELFE-222-Search-View-user-profile/?path=/docs/data-display-user-profile--docs

**Comment by dptelicent** [comment]
> The pop up seems to be opening in the incorrect place here. It should be opening underneath https://telicent-oss.github.io/telicent-ds/TELFE-222-Search-View-user-profile/?path=/docs/data-display-user-profile--docs

Fixed.

File: src/v1/components/data-display/UserProfile/UserProfile.tsx [hunk 1]
```diff
@@ -25,36 +29,41 @@ const UserProfile: React.FC<UserProfileProps> = ({ fullName, children }) => {
...
-        variant="circular-outlined"
-        sx={{ width: 32, height: 32, borderColor: "inherit" }}
-      >
-        <UserIcon color="inherit" />
-      </MUIAvatar>
-      <MUIIconButton
-        aria-label="user-profile"
-        color="inherit"
+    <>
+      <MUIStack
```
Comment (AlecsMissangu) [inline src/v1/components/data-display/UserProfile/UserProfile.tsx:33]: Can we raise a ticket to change the from a stack to a button to improve accessibility. I have just realised there is an additional onClick event.

**Review by AlecsMissangu — APPROVED** [review]

File: src/v1/components/data-display/UserProfile/UserProfile.tsx (hunk 1 above)
Comment (dptelicent) [reply src/v1/components/data-display/UserProfile/UserProfile.tsx:33]: Done: https://telicent.atlassian.net/browse/TELFE-725.

**Pull request merged**